src/OrchardCore.Modules/OrchardCore.Media.Azure/Startup.cs

```diff
@@ -35,26 +44,89 @@ public override void ConfigureServices(IServiceCollection services)
             var containerName = _configuration[$"OrchardCore.Media.Azure:{nameof(MediaBlobStorageOptions.ContainerName)}"];
             if (MediaBlobStorageOptionsCheckFilter.CheckOptions(connectionString, containerName, _logger))
             {
+                // Register a media cache file provider.
+                services.AddSingleton<IMediaCacheFileProvider>(serviceProvider =>
+                {
+                    var hostingEnvironment = serviceProvider.GetRequiredService<IHostingEnvironment>();
+
+                    if (String.IsNullOrWhiteSpace(hostingEnvironment.WebRootPath))
+                    {
+                        throw new Exception("The wwwroot folder for serving cache media files is missing.");
+                    }
+
+                    var mediaOptions = serviceProvider.GetRequiredService<IOptions<MediaOptions>>().Value;
+                    var mediaBlobOptions = serviceProvider.GetRequiredService<IOptions<MediaBlobStorageOptions>>().Value;
+                    var shellOptions = serviceProvider.GetRequiredService<IOptions<ShellOptions>>();
+                    var shellSettings = serviceProvider.GetRequiredService<ShellSettings>();
+                    var logger = serviceProvider.GetRequiredService<ILogger<MediaBlobFileProvider>>();
+
+                    var mediaCachePath = GetMediaCachePath(hostingEnvironment, shellSettings, mediaBlobOptions.AssetsCachePath);
+
+                    if (!Directory.Exists(mediaCachePath))
```

Comment:
Moved the blob media cache to hosting environment `wwwroot/the-tenants-name/ab-cache`
ImageSharp remains as it was, so `wwwroot/is-cache`

Reply:
It's good to have the `AssetsCachePath` configurable but it will not be easy to prevent it from being published (as we do for the IS cache). Is it possible to have at least one hard coded segment?

Reply:
Done. Moved cache to feature, which doesn't know about Azure, and the Azure feature now depends on the Media Cache Feature. Folder name is `ms-cache` (media-store-cache)

@jtkech You are right. I've hardcoded the root folder.

Reply:
👍

Reply:
So, in `OrchardCore.Application.Cms.Core.Targets.props`

    <PropertyGroup>
      <DefaultItemExcludes>$(DefaultItemExcludes);wwwroot\is-cache\**;wwwroot\your-cache\**</DefaultItemExcludes>
    </PropertyGroup>



As i have time i will try to do some more reviewing